etc/systemd/system-generators/zfs-mount-generator.in

```diff
@@ -37,14 +37,17 @@ do_fail() {
 # test if $1 is in space-separated list $2
 is_known() {
   query="$1"
+  OIFS="$IFS"
   IFS=' '
   # protect against special characters
   set -f
   for element in $2 ; do
     if [ "$query" = "$element" ] ; then
+      IFS="$OIFS"
       return 0
     fi
   done
+  IFS="$OIFS"
```

Comment:
I looked a bit more closely at this.  I'm pretty sure `IFS` is only being set here because of us not resetting `IFS` down in `process_line` ("us" is actually me; I'm to blame for that, sorry).  No tabs, newlines, or other whitespace characters should be showing up in `$2`, since that winds up being `$noauto_files`, which is in turn a space-separated list of `$mountfile`s, which are escaped by `systemd-escape`.

So, I think we should just remove the IFS here entirely.

Reply:
IIRC I put it there mostly to formally really enforce the space-separation in case someone threw some funky values in there, but your systemd-escape explanation seems valid to me. Re removing it: What if someone runs the script with a mangled (for example assume empty) IFS